### ISSUE TITLE
When globally installing a tool, a reference to Microsoft.SourceLink.AzureRepos.Git is added causing NU1211 & NU1212

### ISSUE THREAD

**Issue by joslat** [dotnet/sdk#33116]
### Describe the bug
In a Windows11 laptop, with latest .NET SDK, 7.0.302 when I try to install a .NET Tool, somehow the .NET SDK adds a reference to Microsoft.SourceLink.AzureRepos.Git and this triggers two issues: NU1211 & NU1212.
 error NU1211: Project restore must
have a single package reference(s).
C:\...\Local\Temp\e77043b8-a3d2-4c67-bfcb-40fffd2ba162\restore.csproj : error NU1212: Invalid project-package combination for Microsoft.SourceLink.AzureRepos.Git 1.1.1. DotnetToolReference project style can only contain references of the DotnetTool type
The tool package could not be restored.

### To Reproduce
Essentially to reproduce in my environment, I either install any tool globally or do it locally by:
- `dotnet new tool -n MySampleTool  `
- editing the csproj file to make it a tool:
`
<OutputType>Exe</OutputType>  
<TargetFramework>net7.0</TargetFramework>  
<PackAsTool>true</PackAsTool>  
<ToolCommandName>MySampleTool</ToolCommandName> 
`
- building with `dotnet build`
- creating a NuGet package: `dotnet pack --configuration Release --output ./nupkgs  `
- Installing the package as a  global tool: `dotnet tool install -g --add-source ./nupkgs MySampleTool ` 

That said I am unsure that this happens with due to the .NET SDK 7.0.302 or there is a misconfiguration anywhere... I used the .NET repair tool, updated Visual Studio to the latest version and uninstalled other SDK.

And it keeps happening... any help would be really welcome...

### Further technical details
- Include the output of `dotnet --info`
- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version

**Comment by joslat** [comment]
Not directly in the Temp root and not with that name, or with extension .props or .targets - I also looked for references to Microsoft.SourceLink.AzureRepos.Git. 
The temporal directories created by dotnet tool install seem to be wiped out after the "build".

If it helps I have exported the verbose output of the dotnet tool install to a file, using this instruction:
`dotnet tool install -g -v d --configfile disable_nuget.config --add-source ./nupkgs MySampleTool --version 1.2.3 >> toolInstall3.log`
The disable_nuget.config I have set up at the root of the project and contains:
`<?xml version="1.0" encoding="utf-8"?>  
<configuration>  
  <packageSources>  
    <clear />  
      
	<add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
  </packageSources>  
</configuration> `

I am adding the toolinstall3.log as an attachment.

The first time the "Microsoft.SourceLink.AzureRepos.Git reference appears is in line 805, 
`       Task "GetRestorePackageReferencesTask"
         (in) ProjectUniqueName 'C:\Users\josla\AppData\Local\Temp\978712c6-6513-4710-8cac-5d38969f16f1\restore.csproj'
         (in) TargetFrameworks 'net7.0'
         (in) PackageReferences 'Microsoft.SourceLink.AzureRepos.Git;mysampletool'`
which seems to start with the task "CollectCentralPackageVersions" on line 788
And this is what I believe is triggering the two errors.

Any hint on what is (or can be) the issue?

At this point I am thinking on removing Visual Studio, all .NET frameworks...  and performing a clean of any leftover configurations in this regards -and reinstall. Any hint on how to do this process properly? (in a windows 11 machine) 

[toolInstallV2.log](https://github.com/dotnet/sdk/files/11703616/toolInstallV2.log)

**Comment by joslat** [comment]
I've tried those steps in two or more windows 11 pc's and they work flawlessly. no issues.

**Comment by joslat** [comment]
Finally got it!

It was a props file (Directory.Build.Props) hiding in my user's directory root, after a long stream of files... 

Found it by executing a powershell command to find any file containing this string. It took a while but it did the trick.

the script, for if it can help anybody:
`Get-ChildItem -Path .\ -Recurse -Include *.* -File -ErrorAction SilentlyContinue | Select-String -Pattern "Microsoft.SourceLink.AzureRepos.Git" -ErrorAction SilentlyContinue | ForEach-Object { $_.Path } | Out-File -Encoding utf8 -FilePath "c:\search_results.txt"`

I removed that file and others with the same date-time stamp and all seems to work fine now.
